backend/fms_core/services/derived_sample.py

```diff
@@ -0,0 +1,43 @@
+from django.core.exceptions import ValidationError
+from fms_core.models import Biosample, DerivedSample, DerivedBySample, Sample, Container, Process
+
+def get_single_derived_sample_from_sample(sample):
```

Comment:
ugh une fonction de 10 ligne pour remplacer ".first()" ?

Reply:
could be the function you create in the model. for now, I had to repeat that logic more than 5 times in the code..

Reply:
you still have to copy the logic outside unless you want this to crash all around the place ; ). the only original code i see in here is the ".first()"

Reply:
that is if you have a direct link from sample to derived_sample, which is not the case on my branch

Reply:
mmm yeah you are right about that ... well we'll see for another review down the line if it is still required but yeah you are correct about the immediate use ...